roles/pulp_health_check/tasks/main.yml

```diff
@@ -1,19 +1,10 @@
 ---
 - meta: flush_handlers
 
-- name: Get services
-  service_facts:
-
-- name: Listing Pulp services
-  debug:
-    msg: '{{ ansible_facts.services|dict2items|selectattr("key", "contains", "pulp")|list|items2dict }}'
-
-
-- name: Checking Pulp services
-  fail:
-    msg: "{{ item.value.name }} state: {{ item.value.state }}"
-  when: "item.value.name != 'pulpcore-worker@.service' and item.value.state != 'running'"
-  with_dict: '{{ ansible_facts.services|dict2items|selectattr("key", "contains", "pulp")|list|items2dict }}'
+- name: Check if Pulp services have started yet (loop)
+  include_tasks: service_check.yml
+  when: not __service_check_success
+  with_sequence: start=1 end=50
```

Comment:
maybe it is too much, the test is taking too long

Reply:
This is in draft state. I know how to fix it, it will just take me a while.